src/legacy/core_plugins/kibana/public/discover/angular/doc_table/components/table_row.ts

```diff
@@ -19,32 +19,29 @@
 
 import _ from 'lodash';
 import $ from 'jquery';
+// @ts-ignore
 import rison from 'rison-node';
 import '../../doc_viewer';
+// @ts-ignore
 import { noWhiteSpace } from '../../../../../common/utils/no_white_space';
 
 import openRowHtml from './table_row/open.html';
 import detailsHtml from './table_row/details.html';
-import { getServices } from '../../../kibana_services';
+
 import { dispatchRenderComplete } from '../../../../../../../../plugins/kibana_utils/public';
 import cellTemplateHtml from '../components/table_row/cell.html';
 import truncateByHeightTemplateHtml from '../components/table_row/truncate_by_height.html';
 import { esFilters } from '../../../../../../../../plugins/data/public';
 
-const module = getServices().uiModules.get('app/discover');
-
 // guesstimate at the minimum number of chars wide cells in the table should be
 const MIN_LINE_LENGTH = 20;
 
-/**
- * kbnTableRow directive
- *
- * Display a row in the table
- * ```
- * <tr ng-repeat="row in rows" kbn-table-row="row"></tr>
- * ```
- */
-module.directive('kbnTableRow', function ($compile, $httpParamSerializer, kbnUrl, config) {
+export function createTableRowDirective(
+  $compile: any,
+  $httpParamSerializer: any,
+  kbnUrl: any,
+  config: any
```

Comment:
We should have a type something similar like `UiSettingsClient` for that.

Reply:
merci! changed that